src/VisualStudio/Core/Def/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -122,9 +131,10 @@ private VisualStudioRemoteHostClientProvider(
 
             var configuration =
                 _globalOptions.GetOption(RemoteHostOptionsStorage.OOPServerGCFeatureFlag) ? RemoteProcessConfiguration.ServerGC : 0;
+            var localSettingsDirectory = new ShellSettingsManager(_serviceProvider).GetApplicationDataFolder(ApplicationDataFolder.LocalSettings);
 
             // VS AsyncLazy does not currently support cancellation:
-            var client = await ServiceHubRemoteHostClient.CreateAsync(Services, configuration, _listenerProvider, serviceBroker, _callbackDispatchers, CancellationToken.None).ConfigureAwait(false);
+            var client = await ServiceHubRemoteHostClient.CreateAsync(Services, configuration, localSettingsDirectory, _listenerProvider, serviceBroker, _callbackDispatchers, CancellationToken.None).ConfigureAwait(false);
```

Comment:
interesting.  we have an IThreadingContext, but we're not using the disposal token ehre.  feels... wrong.  though i see you're not changing that.